src/java/org/apache/cassandra/index/sai/disk/v1/postings/PostingsWriter.java

```diff
@@ -123,7 +128,6 @@ public PostingsWriter(IndexDescriptor indexDescriptor, IndexContext indexContext
     private PostingsWriter(IndexOutput dataOutput, int blockSize) throws IOException
     {
         assert dataOutput instanceof IndexOutputWriter;
```

Comment:
Any reason not to remove this `assert` and just change the type of `dataOutput` to `IndexOutputWriter`? I tried it, and after a couple minor changes up through to `NumericIndexWriter` (where `openPerIndexOutput()` does give us an `IndexOutputWriter ) it seems to work fine.

Reply:
I nearly removed this when I removed the spamming log statement below it. I have made this change now.